CONTRIBUTING.md

```diff
@@ -0,0 +1,188 @@
+    - [Documentation](#documentation)
+    - [Writing tests](#writing-tests)
+
+## Code Contributions
+
+#### Step 1: Clone
+
+Clone the project on [GitHub](git@github.com:Green-Software-Foundation/ief.git)
+   
+   ```bash
+   $ git clone git@github.com:Green-Software-Foundation/ief.git
+   $ cd ief
+   ```
+   
+For developing new features and bug fixes, the development branch should be pulled and built upon.
+
+#### Step 2: Branch
+
+Create new branch from main (`staging` or `master`), which will contain your new feature, fix or change. 
+
+    ```bash
+    $ git checkout -b <topic-branch-name>
+    ```
+
+#### Step 3: Commit
+
+Make sure git knows your name and email address:
+
+   ```bash
+   $ git config --global user.name "Example User"
+   $ git config --global user.email "user@example.com"
+   ```
+
+Commiting multiple files with changes on multiple resources is strongly restricted.
+Commit message should clearly describe on which resource changes are made.
+
+Add and commit:
+
+   ```bash
+   $ git add my/changed/files
+   $ git commit
+   ```  
+    
+Commit your changes in logical chunks. Do not push all changes in one commit!! 
+
+Please adhere to these [Commit message guidelines](#commit-message-guidelines)
+   or your code is unlikely be merged into the main project.
+
+#### Step 4: Sync
+
+Use git pull/merge to synchronize your work with the main repository.
+
+   ```bash
+   $ git pull origin dev
+   ```
+   
+#### Step 5: Push
+
+Push your topic branch:
+
+   ```bash
+   $ git push origin <topic-branch-name>
+   ```
+
+#### Step 6: Pull Request
+
+Open a Pull Request with a clear title and description.
+
+Pull request should pass all CI which are defined, should have at least one approve.
+
+CI can include lint checks, running tests,  and etc.
+
+## Commit message guidelines
+
+The commit message should describe what changed and why.
+
+   1. The first line should:
+       * contain a short description of the change
+       * be 60 characters or less
+       * be prefixed with the name of the changed subsystem
+       * be entirely in lowercase with the exception of proper nouns, acronyms, and the words that refer to code,
+         like function/variable names
+        
+       Examples:
+       
+       ```
+        util: add getInitializedModel method to models.
+        deps: add express package to dependencies.
+        service: refactor get user.
+       ```
+   2. Keep the second line blank. 
+          
+   3. Wrap all other lines at 72 columns:
+      * describe each line in logical chunks
+      * start each line with: space hyphen space ( - ...)
+      * be entirely in lowercase with the exception of proper nouns, acronyms, and the words that refer to code,
+        like function/variable names
+      
+      Examples:
+      
+      ```    
+        - remove deprecated logger
+        - refactor some method
+        - add JSDoc on existing function
+      ```
+## Coding guidelines
```

Comment:
can also add a note about adhering to the specification - PRs generally won't be accepted if they break from spec